db/fmp_to_postgres.py

```diff
@@ -0,0 +1,148 @@
+            new_df = new_df[new_df["fk_med_key"].notna()]
+            new_df.drop_duplicates(subset=["pk_meds_dispensed","fk_visit_id", "fk_med_key"], inplace=True)
+            print(f"Table ({table_name}) lost ({len(df) - len(new_df)}) entries")
+            df = new_df
+        elif table_name == "diagnoses_made":
+            new_df = df[df["pk_diagnoses_made"].notna()]
+            new_df = new_df[new_df["fk_visit_id"].notna()]
+            new_df = new_df[new_df["fk_diagnosis_key"].notna()]
+            new_df.drop_duplicates(subset=["pk_diagnoses_made", "fk_visit_id", "fk_diagnosis_key"], inplace=True)
+            print(f"Table ({table_name}) lost ({len(df) - len(new_df)}) entries")
+            df = new_df
+        elif table_name == "settings":
+            df['pk_settings_id'] = df.index 
+        elif table_name == "patients": 
+            df.drop(["face_photo"], axis=1, inplace=True)
+            query = "SELECT GetAs(face_photo, 'JPEG') FROM patients"
+            result = fmp.cursor.execute(query)
+            data = result.fetchall()
+            data = [item[0] for item in data]
+            df["face_photo"] = data
+
+            
+        psql.write_table(df, table_name, if_exists="replace")
+
+        pk_field, fk_fields = get_pk_and_fk_from_table(fmp, df, table_name)
+        all_data[table_name] = (pk_field, fk_fields)
+    
+    filename = "table_data.npy"
+    save_keys(filename, all_data)
+    
+
+    filename = "table_data.npy"
+    all_data = load_dict(filename)
+
+    for table_name in all_data.keys():
+        if table_name not in fmp._join_tables:
+            pk_field, _ = all_data[table_name]
+            if pk_field is not None:
+                psql.execute_command(f"ALTER TABLE {table_name} ALTER COLUMN {pk_field} TYPE numeric USING {pk_field}::numeric;")
+                psql.execute_command(f"ALTER TABLE {table_name} ADD PRIMARY KEY ({pk_field});")
+        else:
+            if table_name == "meds_dispensed":
+                for pk_field in ["pk_meds_dispensed", "fk_visit_id"]:
+                    psql.execute_command(f"ALTER TABLE {table_name} ALTER COLUMN {pk_field} TYPE numeric USING {pk_field}::numeric;")
+                psql.execute_command(f"ALTER TABLE {table_name} ADD PRIMARY KEY (pk_meds_dispensed, fk_visit_id, fk_med_key);")
+            elif table_name == "diagnoses_made":
+                for pk_field in ["pk_diagnoses_made", "fk_visit_id"]:
+                    psql.execute_command(f"ALTER TABLE {table_name} ALTER COLUMN {pk_field} TYPE numeric USING {pk_field}::numeric;")
+                psql.execute_command(f"ALTER TABLE {table_name} ADD PRIMARY KEY (pk_diagnoses_made, fk_visit_id, fk_diagnosis_key);")
+
+
+    for table_name in all_data.keys():
+        _, fk_fields = all_data[table_name]
+
+        for fk_field, fk_table in fk_fields:
+            pk_fk_table = all_data[fk_table][0]
+            # POTENTIALLY SUS: ASSUMES WE ARE ALWAYS LINKING TO THE PK OF THE TABLE
```

Comment:
Can we remove `POTENTIALLY SUS:` and just leave the comment

Reply:
will do

Reply:
will do

Reply:
fixed https://github.com/Haiti-Distributed-Healthcare-System/hdhs/pull/64/commits/3dbd4d583fc9b9251c272e8b7250721ed63a56b4